rllib/policy/torch_policy.py

```diff
@@ -467,12 +471,18 @@ def learn_on_batch(self, postprocessed_batch: SampleBatch) -> Dict[str, TensorTy
 
         if self.model:
             fetches["model"] = self.model.metrics()
+
         fetches.update(
             {
                 "custom_metrics": learn_stats,
                 NUM_AGENT_STEPS_TRAINED: postprocessed_batch.count,
+                NUM_GRAD_UPDATES_LIFETIME: self.num_grad_updates + 1,
+                DIFF_NUM_GRAD_UPDATES_VS_SAMPLER_POLICY: (
+                    self.num_grad_updates - (postprocessed_batch.num_grad_updates or 0)
+                ),
             }
         )
+        self.num_grad_updates += 1
```

Comment:
move this to before line 75? then compute `DIFF_NUM_GRAD_UPDATES_VS_SAMPLER_POLICY` as `self.num_grad_updates - (postprocessed_batch.num_grad_updates or 0) - 1`

It makes more sense from a readability perspective, because the assumption is that the most recent gradient computation has already happened. 

alternatively, make a new  temporary var called `previous_num_grad_updates` which is `self.num_grad_updates` and use that for computing `DIFF_NUM_GRAD_UPDATES_VS_SAMPLER_POLICY`

Reply:
You are right, we should increase the counter as soon as we do the actual update. Fixed.